model/pdata/common.go

```diff
@@ -193,6 +206,14 @@ func (a AttributeValue) ArrayVal() AnyValueArray {
 	return newAnyValueArray(&arr.Values)
 }
 
+// BytesVal returns the []byte value associated with this AttributeValue.
+// If the Type() is not AttributeValueTypeBytes then returns false.
+// Calling this function on zero-initialized AttributeValue will cause a panic.
+// Modifying the returned []byte in-place is strongly discouraged.
```

Comment:
let's use "forbidden" here as well, be more strict initially then we can relax if needed.